### PR TITLE
Avoid unnecessary initialization

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Comment by dalg24** [comment]
Note that I did not bother propagating the `ViewAllocateWithoutInitializing` to `Details::DistributedSearchTreeImpl` at this point.

**Comment by dalg24** [comment]
And that they are no tests for `reallocWithoutInitializing()` nor `cloneWithoutInitializingNorCopying()`.

File: packages/Search/src/details/DTK_DetailsTreeConstruction_def.hpp [hunk 1]
```diff
@@ -185,8 +187,10 @@ class CalculateBoundingBoxesFunctor
             if ( Kokkos::atomic_compare_exchange_strong(
                      &_flags( node - _root ), 0, 1 ) )
                 break;
-            for ( Node *child : {node->children.first, node->children.second} )
-                expand( node->bounding_box, child->bounding_box );
+
+            node->bounding_box = node->children.first->bounding_box;
```
Comment (aprokop) [inline packages/Search/src/details/DTK_DetailsTreeConstruction_def.hpp]: Add a comment here that `node->bounding_box` is not initialized before this code.

File: packages/Search/src/details/DTK_DetailsUtils.hpp [hunk 2]
```diff
@@ -217,6 +217,41 @@ accumulate( ViewType const &v, typename ViewType::non_const_value_type init )
...
+{
+    v = View( Kokkos::ViewAllocateWithoutInitializing( v.label() ), n0, n1, n2,
+              n3, n4, n5, n6, n7 );
+}
+
+template <typename View>
+typename View::non_const_type cloneWithoutInitializingNorCopying( View &v )
+{
+    return typename View::non_const_type(
+        Kokkos::ViewAllocateWithoutInitializing( v.label() ), v.layout() );
```
Comment (aprokop) [inline packages/Search/src/details/DTK_DetailsUtils.hpp]: Not sure if the label should be modified to include "clone"

**Comment by dalg24** [comment]
`--values=10000000 --queries=1000000`

Cuda (P100)  | eaeee1b (current master) | 97fbc0a (this pull request)
------------ | -------- | --------
construction | 0.277002 | 0.228262
kNN          | 0.388554 | 0.38362
radius       | 0.260069 | 0.255035


OpenMP (16 threads)  | eaeee1b (current master) | 97fbc0a (this pull request)
------------ | -------- | --------
construction | 0.350321 | 0.333785
kNN          | 0.365086 | 0.365617
radius       | 0.38013 | 0.38306


Serial  | eaeee1b (current master) | 97fbc0a (this pull request)
------------ | -------- | --------
construction | 3.50028 | 3.32917
kNN          | 4.27742 | 4.35997
radius       | 4.44639 | 4.49315

File: packages/Search/src/details/DTK_DetailsUtils.hpp (hunk 2 above)
Comment (dalg24) [reply packages/Search/src/details/DTK_DetailsUtils.hpp]: Nah

**Pull request merged**